src/consensus/params.cpp

```diff
@@ -12,6 +12,46 @@
 #include "util/match.h"
 
 namespace Consensus {
+    /**
+     * General information about each funding stream.
+     * Ordered by Consensus::FundingStreamIndex.
+     */
+    constexpr struct FSInfo FundingStreamInfo[Consensus::MAX_FUNDING_STREAMS] = {
+        {
+            .recipient = "Electric Coin Company",
+            .specification = "https://zips.z.cash/zip-0214",
+            .valueNumerator = 7,
+            .valueDenominator = 100,
+        },
+        {
+            .recipient = "Zcash Foundation",
+            .specification = "https://zips.z.cash/zip-0214",
+            .valueNumerator = 5,
+            .valueDenominator = 100,
+        },
+        {
+            .recipient = "Major Grants",
+            .specification = "https://zips.z.cash/zip-0214",
+            .valueNumerator = 8,
+            .valueDenominator = 100,
+        },
+        {
+            .recipient = "Lockbox",
+            .specification = "https://zips.z.cash/draft-nuttycom-funding-allocation",
+            .valueNumerator = 20,
+            .valueDenominator = 100,
+        }
+    };
+    static constexpr bool validateFundingStreamInfo(uint32_t idx) {
+        return (idx >= Consensus::MAX_FUNDING_STREAMS || (
+            FundingStreamInfo[idx].valueNumerator < FundingStreamInfo[idx].valueDenominator &&
+            FundingStreamInfo[idx].valueNumerator < (INT64_MAX / MAX_MONEY) &&
+            validateFundingStreamInfo(idx + 1)));
+    }
+    static_assert(
+        validateFundingStreamInfo(Consensus::FIRST_FUNDING_STREAM),
+        "Invalid FundingStreamInfo");
```

Comment:
It's possible to have a `static_assert` that checks all members of a list in O(1) (and not particularly complicated) code by using recursion, along the lines of https://stackoverflow.com/a/33987081/393146 .